src/common/TracepointProvider.cc

```diff
@@ -39,6 +39,7 @@ void TracepointProvider::verify_config(const struct md_config_t *conf) {
   void *handle = dlopen(m_library.c_str(), RTLD_NOW);
   if (handle != NULL) {
     m_enabled = true;
+    free(handle);
```

Comment:
@xiexingguo that would unload the shared object and I'm not sure that's the intention here? You're right though that free is probably not the way to go in hindsight. @dillaman could we get your opinion here?